From dd1602e306fec366280f5953c5e1b553e3d9672a Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Tue, 3 Sep 2024 17:10:10 -0400
Subject: [PATCH 2/3] firefox: set MOZ_* variables for legacy profiles.ini

This fixes app startup for installations that are not using
`wrapFirefox` (e.g. app from `homebrew`).

These variables are already set for nixpkgs firefox using wrapFirefox
wrapper.

This is most helpful for Darwin users. But it will help with an
(unusual) scenario where Home Manager is used in a non-NixOS environment
with the app managed by another package manager, or installed directly
from Mozilla builds.

Closes: #5717
---
 modules/programs/firefox/mkFirefoxModule.nix | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/programs/firefox/mkFirefoxModule.nix b/modules/programs/firefox/mkFirefoxModule.nix
index f22af019..69ef3c37 100644
--- a/modules/programs/firefox/mkFirefoxModule.nix
+++ b/modules/programs/firefox/mkFirefoxModule.nix
@@ -1012,6 +1012,12 @@ in {
           force = true;
         };
     }));
+
+    # Mimic nixpkgs package environment for read-only profiles.ini management
+    home.sessionVariables = {
+      MOZ_LEGACY_PROFILES = 1;
+      MOZ_ALLOW_DOWNGRADE = 1;
+    };
   } // setAttrByPath modulePath { finalPackage = wrapPackage cfg.package; });
 }
 
-- 
2.46.0

